load-gen/src/main/java/io/smartcat/data/loader/api/RateGenerator.java

```diff
@@ -0,0 +1,6 @@
+package io.smartcat.data.loader.api;
+
+public interface RateGenerator {
+
+    long getRate(long prevInstant, long currentInstant);
```

Comment:
Yes, if you have RateGenerator implementation as: getRate(...) { return 1000; }
Then you need to have notion of time somewhere in code which will ensure this limit.
It seems that it would be good idea to separate rateGeneartor from time component. But I haven't figured out yet how to do it without introducing tmer tick every 1ms.